### PR TITLE
[FLOC-4190] Fix a bug in the initialize-release tool

### PR DESCRIPTION
**Pull request by wallrj**
Fixes: https://clusterhq.atlassian.net/browse/FLOC-4190

### CODE REVIEW

File: admin/release.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- test-case-name: admin.test.test_release -*-
+# -*- test-case-name: admin.test.test_release,admin.functional.test_release -*-  # noqa
```
Comment (sarum90) [inline admin/release.py]: :question: I don't have emacs or emacs plugins that parse this, so I'll have to trust your that this is the right form. (Just fyi, I'm sure it's right, but I can't verify)

File: admin/release.py [hunk 2]
```diff
@@ -879,7 +875,7 @@ def initialize_release(version, path, top_level):
         os.environ["LDFLAGS"] = '-L{}/lib" CFLAGS="-I{}/include'.format(
             brew_openssl, brew_openssl)
     check_call(
-        ["pip install -e .[dev]"], shell=True,
+        ["pip", "install", "-e" ".[dev]"],
```
Comment (sarum90) [inline admin/release.py]: missing comma after -e I think? Might still work cause single letter args sometimes parse all characters after the first as the arg parameter. If that's the case and it is intentional, prefer to delete the space between the strings.

File: admin/functional/test_release.py [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        self.expectThat(result.status, Equals(0))
+        self.expectThat(result.output, Contains('Usage: initialize-release'))
+
+    def test_run(self):
+        """
+        The script creates a new release version branch in a new working
+        directory and prints some shell commands to ``stdout``.
+        """
+        expected_version = "9.9.9"
+        script_path = self.make_temporary_directory()
```
Comment (sarum90) [inline admin/functional/test_release.py]: :question: Just FYI I don't think this cleans up the directory, just puts it in the temp folder. probably okay, but thought you should know.

File: admin/functional/test_release.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        repo_path = script_path.sibling(
+            'flocker-release-{}'.format(expected_version)
+        )
+        result = run_process(
+            [INITIALIZE_RELEASE_PATH.path,
```
Comment (sarum90) [inline admin/functional/test_release.py]: :question: I don't remember off hand and should be getting ready to go here pretty soon. Does this also make the virtualenv for you? If so we might want to add cleanup to this test. If not, fine as is.

**Comment by sarum90** [comment]
Looks good, thanks for maintaining these scripts.

One real comment to address before merging.

**Comment by tomprince** [comment]
I'd strongly suggest, instead of fixing the script, deleting it, and replacing the instructions to use it with prose instructions to create a new work-copy and virtualenv.

**Comment by tomprince** [comment]
To expand on that:
- less code to maintain
- when releases get automated, the jenkins abstraction around getting a clean environment will be used, rather than this.

File: admin/release.py (hunk 1 above)
Comment (wallrj) [inline admin/release.py]: Fair point. I tested it in emacs, but this is also how `trial  --testmodule=admin/release.py` works. We use that feature for running our flocker.node.agent functional tests.

File: admin/release.py (hunk 2 above)
Comment (wallrj) [inline admin/release.py]: Fixed.

File: admin/functional/test_release.py (hunk 3 above)
Comment (wallrj) [inline admin/functional/test_release.py]: That's what I want. So I can see what the script created if the test fails.

File: admin/functional/test_release.py (hunk 4 above)
Comment (wallrj) [inline admin/functional/test_release.py]: It does create a virtualenv, but inside the newly cloned working copy.
```
ls -la _trial_temp/admin.functional.test_release/InitializeReleaseScriptTests/test_run/flocker-release-9.9.9/flocker-9.9.9/bin/activate
```

So no need to clean up.

**Comment by wallrj** [comment]
> I'd strongly suggest, instead of fixing the script, deleting it, and replacing the instructions to use it with prose instructions to create a new work-copy and virtualenv.

I kind of agree. But I notice that the command also handles some stuff to do with installing OpenSSL on the OSX platform. I don't want to have to document that right now.
This command obviously worked for @zubron  during the last few releases, so I'll leave it for now and it can be deleted if / when we do fully automate the release process.

**Comment by zubron** [comment]
@wallrj: I ended up only doing one release which used this script. I fixed one bug in it during that release [FLOC-3842](https://clusterhq.atlassian.net/browse/FLOC-3842) and then I encountered another bug that didn't affect me until later in the process. See [FLOC-3918](https://clusterhq.atlassian.net/browse/FLOC-3918).

**Comment by wallrj** [comment]
@zubron Oh, right :frowning: so this script hasn't got a great track record then. Perhaps next release I'll take another look and see if it can be deleted and documented instead. (as @tomprince suggested)

**Pull request merged**